.eslintrc.js

```diff
@@ -152,8 +152,21 @@ module.exports = {
 			},
 			{
 				selector: 'CallExpression[callee.object.name="page"][callee.property.name="waitFor"]',
-				message: 'Prefer page.waitForSelector instead.'
-			}
+				message: 'Prefer page.waitForSelector instead.',
+			},
+			{
+				// The <DotTip> component uses the `id` prop for something that does not require an
+				// instanceId; maybe we should change its key.
+				selector: 'JSXOpeningElement[name.name!="DotTip"] JSXAttribute[name.name="id"][value.type="Literal"]',
+				message: 'Do not use string literals for IDs; use withInstanceId instead.',
+			},
+			{
+				// Discourage the usage of `Math.random()` as it's a code smell
+				// for UUID generation, for which we already have a higher-order
+				// component: `withInstanceId`.
+				selector: 'CallExpression[callee.object.name="Math"][callee.property.name="random"]',
```

Comment:
Might we want to contextualize this? To `id`?

https://astexplorer.net/#/gist/4f09c71cd79ebaca02a540aa8c2c4f9b/a3d4f17bf669d19d38b50a06dd1d50d8a9526360

Or is it intentional to cover more usage than just in `id`?

Reply:
It's intentional because we don't use it anywhere in the code base at the moment, and in the case of the Datepicker PR (#7621), it wasn't directly assigned to the `id` prop. It was assigned to a `state` variable and then used later, so I'm not sure it'd have been caught.

When do we ever really want to be using `Math.random()` anyhow? 😉